pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -85,8 +84,8 @@ def __init__(
         tpu_cores,
         ipus,
         distributed_backend,
-        auto_select_gpus,
         gpus,
+        gpu_ids,
```

Comment:
that is the same as `gpus` above

Reply:
`gpus` is User input. `gpu_ids` is a list of parsed gpu ids.

Reply:
just feel very confused...

Reply:
@Borda The reason we have to do this, is because we have Trainer properties based on `gpus` and `gpu_ids`. So, they both need to exist in Acc connector. 
Have added a comment!

Reply:
let's definitely not confuse both together, the explicit differentiation is better than combining the two

Reply:
but on the other hand, we have only one `tpu_cores` right?

Reply:
@Borda Yup, there's a reason for that. You can't call `xm.xla_device()` to fetch device info outside the spawn process.
This is a refactor PR, with no extra unintended changes.